src/main/kotlin/com/demonwav/mcdev/platform/fabric/creator/FabricProjectCreator.kt

```diff
@@ -0,0 +1,440 @@
+        indicator.text2 = null
+    }
+}
+
+class LicenseStep(
+    private val project: Project,
+    private val rootDirectory: Path,
+    private val license: License,
+    private val config: FabricProjectConfig
+) : CreatorStep {
+
+    override fun runStep(indicator: ProgressIndicator) {
+        val licenseFile = rootDirectory.resolve("LICENSE")
+
+        val fileText = FabricTemplate.applyLicenseTemplate(project, license, config)
+
+        Files.write(
+            licenseFile,
+            fileText.toByteArray(Charsets.UTF_8),
+            StandardOpenOption.CREATE,
+            StandardOpenOption.WRITE,
+            StandardOpenOption.TRUNCATE_EXISTING
+        )
+    }
+}
+
+class FabricModJsonStep(
+    private val project: Project,
+    private val buildSystem: BuildSystem,
+    private val config: FabricProjectConfig
+) : CreatorStep {
+
+    override fun runStep(indicator: ProgressIndicator) {
+        val text = FabricTemplate.applyFabricModJsonTemplate(project, buildSystem, config)
+        val dir = buildSystem.dirsOrError.resourceDirectory
+
+        indicator.text = "Indexing"
+
+        project.runWriteTaskInSmartMode {
+            indicator.text = "Creating 'fabric.mod.json'"
+
+            val file = PsiFileFactory.getInstance(project).createFileFromText(JsonLanguage.INSTANCE, text)
+            file.runWriteAction {
+                val jsonFile = file as JsonFile
+                val json = jsonFile.topLevelValue as? JsonObject ?: return@runWriteAction
+                val generator = JsonElementGenerator(project)
+
+                (json.findProperty("authors")?.value as? JsonArray)?.let { authorsArray ->
+                    for (i in config.authors.indices) {
+                        if (i != 0) {
+                            authorsArray.addBefore(generator.createComma(), authorsArray.lastChild)
+                        }
+                        authorsArray.addBefore(generator.createStringLiteral(config.authors[i]), authorsArray.lastChild)
+                    }
+                }
+
+                (json.findProperty("contact")?.value as? JsonObject)?.let { contactObject ->
+                    val properties = mutableListOf<Pair<String, String>>()
+                    val website = config.website
+                    if (!website.isNullOrBlank()) {
+                        properties += "website" to website
+                    }
+                    val repo = config.modRepo
+                    if (!repo.isNullOrBlank()) {
+                        properties += "repo" to repo
+                    }
+                    for (i in properties.indices) {
+                        if (i != 0) {
+                            contactObject.addBefore(generator.createComma(), contactObject.lastChild)
+                        }
+                        val key = StringUtil.escapeStringCharacters(properties[i].first)
+                        val value = "\"${StringUtil.escapeStringCharacters(properties[i].second)}\""
```

Comment:
I think it would be simpler with a concatenation.

```suggestion
                        val value = '"' + StringUtil.escapeStringCharacters(properties[i].second) + '"'
```